libkbfs/block_retrieval_worker.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"io"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+// blockRetrievalWorker processes blockRetrievalQueue requests
+type blockRetrievalWorker struct {
+	blockGetter
+	stopCh chan struct{}
+	queue  *blockRetrievalQueue
+}
+
+// run runs the worker loop until Shutdown is called
+func (brw *blockRetrievalWorker) run() {
+	for {
+		select {
+		case <-brw.stopCh:
+			return
+		default:
+			brw.HandleRequest()
```

Comment:
I agree, although the error is only relevant when it's `io.EOF`. Otherwise it exists purely to be handled in `FinalizeRequest`. I'll add a comment to clarify.